0002-capture-matching-next-steps.md

```diff
@@ -0,0 +1,115 @@
+* Improvement to capture matching efficiency and accuracy offered by enhancement
+* Stability of enhancement across diverse use cases
+* Ease of use for affected users
+
+## Considered Options
+
+1. Implement mobile application side image collection improvements, including enforced phone rotation, enforced compass bearing, and perhaps others.
+2. Research feasibility of leveraging autocorrelation between GPS measurements to improve selection of candidates for location based matching of captures.
+3. Create experimental manual capture matching UI that offers manual spatial analysis relevant to disambiguating matching captures.
+4. Research approaches to image analysis that can produce a similarity metric to improve selection of candidates for location based matching of captures.
+5. Attempt to train models using machine learning based on existing database of capture images to identify repeat captures.
+6. Incorporate assigned species information during capture matching.
+7. Create curated training sets of matched trees for to support training of machine learning based models.
+<!--
+## Decision Outcome
+
+Chosen option: "{option 1}", because {justification. e.g., only option, which meets k.o. criterion decision driver | which resolves force {force} | … | comes out best (see below)}.
+
+### Positive Consequences <!-- optional -->
+
+* {e.g., improvement of quality attribute satisfaction, follow-up decisions required, …}
+* …
+
+### Negative Consequences <!-- optional -->
+
+* {e.g., compromising quality attribute, follow-up decisions required, …}
+* …
+-->
+
+## Pros and Cons of the Options
+
+### 1. Implement mobile application side image collection improvements, including enforced phone rotation, enforced compass bearing, and perhaps others.
+
+Currently, images captured on the mobile application do not have specific criteria guiding capture.  Thus, they exist at a variety of camera rotation angles and compass bearings, even for the same tree being repeat captured.  This requires a manual operator to consider images of the same tree that almost always exist at different rotations and focal distances, greatly enhancing the difficulty of an already challenging manual task.
+
+Constraints can be implemented on the mobile side to ameliorate this situation.  For instance, camera rotation around the x axis could be constrained to 0, 45, and 90 degrees.  Additionally, compass bearing could be constrained such that captured photos are always north facing.  Implementing a constraint on focal distance is also possible, as has been identified in other apps.  
+
+* Good, because the images captured will have more constitant parameters.
+* Good, because this only affect the mobile application and does not require data research or algorithm development.
+* Bad, because the mobile application will become more complex for our low technical literacy users, and we will have to invest careful UX/UI development time to make these features easily usable.
+
```

Comment:
Bad, because some trees may be awkward or impossible to photograph from specific positions.

Reply:
Bad, because any increase in **tracking time per tree** will negatively impact our target users.

Reply:
@Davidezrajay Can you expand here on your thoughts on minor increases in tracking time per tree vs absolute feasibility of capture matching?  Which one takes higher priority?

Reply:
They are both critical. It is a fine line between time and usable data. 

**Time:** If people/orgs won't use the system because it takes too long/is too expensive to track, there will be no data. 

**Data:** If we collect bad/inaccurate data, it will be worth less or worthless.

Prioritize tracking time as it is extremely critical for adoption on the ground.  I recommend picking a usable threshold of an average of 5 seconds per capture. And working around that to get the most accurate data possible.

History - The fastest the app has been has been about 3 seconds, with an average gps accuracy of 10m and an excessive amount of worthless data. The original app had a `setting_id` that allowed us to change the accuracy requirement based on the organizations need. Users could set the required accuracy from something like 2 meters to like 1km.

Reply:
Understood, but unfortunately I have to strong disagree.   

We studied the GPS error on a suite of phones, and found that unless we wait for GPS convergence, the location data is garbage, bad enough to make repeat monitoring (aka capture matching impossible).   It is imperative that we wait for a solid GPS convergence.  If we do not have GPS convergence, none of the approaches discussed in this ADR that use GPS are possible - this is what we found by doing grid based studies and visual analysis.

On good phones, convergence will happen faster, on poor phones it can take more time or lead to a timeout condition.  If a planting organization wants a fast and accurate experience, they do have the option to invest in higher quality phones.

Currently convergence timeout is set to 20 seconds here, we could explore criteria for adjusting this default. https://github.com/Greenstand/treetracker-android/blob/4716e6b575f099e7d68efa666c184f605d12e75b/app/src/main/java/org/greenstand/android/TreeTracker/models/Configuration.kt#L36

Reply:
It is true, we have gotten some complaints about the old app being "better" because it was faster. 
I agree with Zaven focusing on the best possible convergence AND we need to assure the following 

1. The longer waiting time needs to be clearly communicated through the app. At least the organizations need to be totally aware of added waiting time. IE good place to start publishing this, is the Playstore which currently does not have that clause.(i just opened a ticket on this https://github.com/Greenstand/treetracker-android/issues/982) 
2. The waiting time has to have some kind of progress that is visible (i believe https://github.com/Greenstand/treetracker-android/issues/970 is going to address this) - urgency just increased here ... 
3. Possibly have a switch with the option of "waiting for convergence" vs. "fast and useless date"

It is true that without the capture matching tool the convergence is secondary but the biggest fundamental change in the system during the domain model migration is around the tree and the recaptures so matching them has become priority and without accuracy this core functionality will only get further out of reach.

Reply:
As per @ZavenArra' suggestion, I've opened this issue to look at optimizing the convergence. https://github.com/Greenstand/overview-archived/issues/125

We are currently getting feedback on current tracking times being too long.